meerk40t/core/elements.py

```diff
@@ -5540,6 +5541,79 @@ def contains(box, x, y=None):
+        :param elements: list of elements to classify.
+        :param operations: operations list to classify into.
+        :param add_op_function: function to add a new operation, because of a lack of classification options.
+        :return:
+        """
+        if elements is None:
+            return
+        reverse = self.context.classify_reverse
+        if reverse:
+            elements = reversed(elements)
+        if operations is None:
+            operations = list(self.ops())
+        if add_op_function is None:
+            add_op_function = self.add_op
+        for element in elements:
+            was_classified = False
+            if hasattr(element, "operation"):
+                add_op_function(element)
+                continue
+            if element is None:
+                continue
+            for op in operations:
+                if op.operation == "Raster":
+                    if element.stroke is not None and op.color == abs(element.stroke):
+                        op.add(element, type="opnode")
+                        was_classified = True
+                    elif isinstance(element, SVGImage):
+                        op.add(element, type="opnode")
+                        was_classified = True
+                    elif isinstance(element, SVGText):
+                        op.add(element)
+                        was_classified = True
+                    elif element.fill is not None and element.fill.argb is not None:
+                        op.add(element, type="opnode")
+                        was_classified = True
+                elif (
+                    op.operation in ("Engrave", "Cut")
+                    and element.stroke is not None
+                    and op.color == abs(element.stroke)
+                ):
+                    op.add(element, type="opnode")
+                    was_classified = True
+                elif op.operation == "Image" and isinstance(element, SVGImage):
+                    op.add(element, type="opnode")
+                    was_classified = True
+                    break  # May only classify in one image operation.
+                elif (
+                    op.operation == "Dots"
+                    and isinstance(element, Path)
+                    and len(element) == 2
+                    and isinstance(element[0], Move)
+                    and isinstance(element[1], Close)
+                ):
+                    op.add(element, type="opnode")
+                    was_classified = True
+                    break  # May only classify in Dots.
+            if not was_classified:
+                if element.stroke is not None and element.stroke.value is not None:
+                    op = LaserOperation(
+                        operation="Engrave", color=element.stroke, speed=35.0
+                    )
+                    add_op_function(op)
+                    op.add(element, type="opnode")
+                    operations.append(op)
```

Comment:
We need to add disabled (output = False) operations for image or Dots or raster if element has not been classified.

Reply:
If there is no existing Dots operation, this will classify dots to an Engrave operation.